src/register/macros.rs

```diff
@@ -134,6 +134,28 @@ macro_rules! write_csr_as_usize_rv32 {
     };
 }
 
+macro_rules! claim_csr_as {
+    ($register:ident, $csr_number:literal) => {
+        /// Claim and return the highest-priority interrupt pending on this hart
+        pub unsafe fn claim() -> $register {
+            match () {
+                #[cfg(riscv)]
```

Comment:
It feels weird to have a check for `riscv` within riscv crate. But, looks like all other macros have it too. Is this for testing purpose?

Reply:
Yeah, there are a couple of tests that can be compiled for the host architecture. I used that to write tests for the APLIC structure to confirm that all of the fields in the struct have the right offset.

Reply:
Kind of tangent, but could those checks be converted to static asserts?  Then the test for the APLIC struct don't ever have to be explicitly run.